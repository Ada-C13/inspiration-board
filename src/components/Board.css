.board {
  display: flex;
  flex-wrap: wrap;
  /* centers board */
  margin-left: auto;
  margin-right: auto;
  align-items: center;
  justify-content: center;
}

.validation-errors-display {
  text-align: center;
  margin-bottom: 2em;
}

.validation-errors-display__list {
  list-style-type: none;
  padding: 0;
}
