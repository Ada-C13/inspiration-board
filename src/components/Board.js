import React, { Component, useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';

import './Board.css';
import Card from './Card';
import NewCardForm from './NewCardForm';
import CARD_DATA from '../data/card-data.json';

const Board = (props) => {
  const BASE_URL = `${props.url}${props.boardName}/cards`

  const[cards, setCards] = useState([]);
  const[errorMessage, setErrorMessage] = useState(null);

  useEffect(() => {
    axios.get(BASE_URL)
    .then((response) => {
      const apiData = response.data;
      console.log(apiData);

      const cardObjects = apiData.map((cardWrapper, i) => {
        return {
          id: cardWrapper.card.id,
          text: cardWrapper.card.text,
          emoji: cardWrapper.card.emoji
        }
      });
      setCards(cardObjects);
      })
      .catch((error) => {
        setErrorMessage(error.message);
      });
  }, []);

  // Wave 3
  const deleteCard = (id) => {
    axios.delete(`https://inspiration-board.herokuapp.com/cards/${id}`)
    .then((response) => {
      const newCards = cards.filter(card => card.id !== id);
      setCards(newCards);
    })
    .catch((error) => {
      setErrorMessage(error.message);
    })
  };

  // Wave 3
  const addCard = (cardInfo) => {
    const nextId = Math.max(...cards.map(card => card.id)) + 1;

    axios.post(BASE_URL, cardInfo)
    .then((response) => {
      const newCards = [...cards];
      const newCard = {
        id: nextId,
        ...cardInfo
      };
      newCards.unshift(newCard);
      setCards(newCards);
    })
    .catch((error) => {
      setErrorMessage(error.message);
    })
  }

  
  const cardComponents = cards.map((card, i) => {
    return (
      <Card 
      id={card.id}
      key={i}
      text={card.text}
      emoji={card.emoji}
      deleteCardCallback={deleteCard}
      />
    );
  });

  return (
    <div>
      {errorMessage && <div><h2>{errorMessage}</h2></div>}
      {cardComponents}
      <NewCardForm 
        addCardCallback={addCard}
      />
    </div>
  )
};


Board.propTypes = {
  url: PropTypes.string.isRequired,
  boardName: PropTypes.string.isRequired
};

export default Board;
