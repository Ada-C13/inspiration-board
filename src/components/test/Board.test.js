import React from 'react';
import Board from '../Board';
import { render, cleanup } from '@testing-library/react'

describe('Board', () => {

  test('that it matches the existing snapshot', () => {
    // Arrange-Act
    const { asFragment } = render(
      <Board
        url="https://inspiration-board.herokuapp.com/boards/"
        boardName="nora-antonia"
      />
    );

    // Assert
    expect(asFragment()).toMatchSnapshot();
    cleanup();
  });
});