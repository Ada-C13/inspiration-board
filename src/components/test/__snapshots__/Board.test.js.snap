// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board that it matches the existing snapshot 1`] = `
<DocumentFragment>
  <article>
    <div />
    <div
      class="board"
    >
      <div
        class="new-card-form"
      >
        <h3
          class="new-card-form__header"
        >
          Leave some inspiration:
        </h3>
        <form
          class="new-card-form__form"
        >
          <input
            class="new-card-form__form-textarea"
            name="text"
            placeholder="An encouraging message"
            type="text"
          />
          <input
            class="new-card-form__form-textarea"
            name="emoji"
            placeholder="emoji name"
            type="text"
          />
          <input
            class="new-card-form__form-button"
            type="submit"
            value="Add note"
          />
        </form>
      </div>
    </div>
  </article>
</DocumentFragment>
`;
